verl/workers/config/reward_model.py

```diff
@@ -0,0 +1,181 @@
+        megatron (Dict[str, Any]): Megatron-specific parallelism settings.
+        load_weight (bool): Whether to load initial weights.
+    """
+
+    strategy: str = "megatron"
+    nccl_timeout: int = 600
+    megatron: McoreEngineConfig = field(default_factory=McoreEngineConfig)
+    load_weight: bool = True
+
+    def validate(self, n_gpus: int):
+        """Validate Megatron reward model configuration with runtime parameters."""
+        super().validate(n_gpus)
+
+
+@dataclass
+class FSDPRewardModelConfig(RewardModelConfig):
+    """Configuration for FSDP-based reward model inference.
+    The inheritance from RewardModelConfig provides all base reward model configuration plus FSDP-specific settings.
+    Args:
+        ulysses_sequence_parallel_size (int): Sequence parallelism size for Ulysses-style model parallelism.
+    """
+
+    strategy: str = "fsdp"
+    ulysses_sequence_parallel_size: int = 1
+
+    def __post_init__(self):
+        """Validate FSDP reward model configuration parameters."""
+        super().__post_init__()
+
+        if self.strategy in {"fsdp", "fsdp2"}:
+            if self.ulysses_sequence_parallel_size > 1:
+                if not self.model.get("use_remove_padding", False):
+                    raise ValueError(
+                        "When using sequence parallelism for reward model, you must enable `use_remove_padding`."
+                    )
+
+    def validate(self, n_gpus: int):
+        """Validate FSDP reward model configuration with runtime parameters."""
+        super().validate(n_gpus)
+
+        if not self.use_dynamic_bsz:
+            sp_size = self.ulysses_sequence_parallel_size
+            if self.micro_batch_size is not None:
+                if self.micro_batch_size * sp_size < n_gpus:
+                    raise ValueError(
+                        f"reward_model.micro_batch_size ({self.micro_batch_size}) * "
+                        f"ulysses_sequence_parallel_size ({sp_size}) must be >= n_gpus ({n_gpus})"
+                    )
+
+
+@dataclass
+class FSDPRewardModelCfg(BaseModelConfig):
+    """FSDP-enabled reward model configuration.
+    Inherits base reward model settings and adds distributed-memory.
+    Args:
+        input_tokenizer (Optional[str]): Input tokenizer. If the reward model's chat template is inconsistent with
+            the policy, we need to first decode to plaintext, then apply the rm's chat_template. Then score with RM.
+            If chat_template are consistent, it can be set to null. set this to null if the chat template is identical.
+        use_shm (bool): Whether to use shared memory for loading the model.
+        use_remove_padding (bool): Use remove-padding optimization (saves compute).
+        use_fused_kernels (bool): Whether to use fused reward kernels for speedup.
+        fsdp_config (FSDPEngineConfig): FSDP-specific configuration block.
+    """
+
+    input_tokenizer: Optional[str] = None
```

Comment:
@vermouth1992 @tongyx361 @PeterSH6 should we replace input_tokenizer with `tokenizer_path`, to make the naming consistent with actor_rollout_ref.model? (breaking change)

Reply:
Maybe fsdp critic.model & reward_model.model should all use the same config class? https://github.com/eric-haibin-lin/verl/blob/main/verl/workers/config/critic.py#L209 @vermouth1992

Reply:
I would actually suggest that we deprecate the support of any form of reward model in the workergroup to favor using reward model as external service (e.g., use sglang/vllm to server)